src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -577,6 +588,17 @@ internal void Save(string filePath, bool isBackup = false, EngineController engi
                 throw (ex);
             }
         }
+        /// <summary>
+        /// This function appends view block to the model json
+        /// </summary>
+        /// <param name="modelData">Workspace Model data in JSON format</param>
+        private JObject AddViewBlockToJSON(JObject modelData)
+        {
+            var token = JToken.Parse(this.ToJson());
```

Comment:
here is a thought around performance... if this is already a JObject, do we ned to convert it to JSON, then reparse it just to add a new property?

Reply:
I did not understand this. `this.tojson` is where the converters are used. what is meant by `convert it to JSON`?

Reply:
yep - you're right, I guess we would need a method on the `workspaceViewModel` which returns a `JObject` as well so we do not need to do this parse every time. I am guessing this parse step is not very fast for 100 nodes.

Reply:
`JObject.Add("string", JToken)`.  You mean't to say `JToken.parse` is slow?

Reply:
I am not sure if `JToken.parse` is slow, but it seems like unnecessary step - I also know we do this same thing for all other serializations. I just meant to highlight that we should be wary of it and maybe profile it with large graphs.

Reply:
The other thing is, if backup runs for every 1 min, then I can read it from backup instead of this step.  
`if backupinterval < 1 min read from backup else serialize`.  
JObject takes JToken as its parameter. One of the other way I could think of is to create an array: array.push(modelJObject) and array.push(viewJObject). then simply convert this array to string or to write to the file.

Reply:
@mjkkirschner I think profiling this for a bunch of large graphs seems like a good idea. I would be a little surprised if the performance impact was significant.

Reply:
>The other thing is, if backup runs for every 1 min, then I can read it from backup instead of this step.

I would not advise doing that unless there are clear performance problems. It adds complexity which increases the chances of something going complicatedly wrong which will just make the data look strange, and be very hard to diagnose.

I suggest finding out if there is an empirical performance problem before trying to do complicated things to fix it. If you like we could report a perf message of how long the heartbeat took and look at the results?

Reply:
@lukechurch Yes. I am not going to touch it. There will be a separate task for that.